API/Repositories/Implementations/PlayerRepository.cs

```diff
@@ -150,28 +162,45 @@ await _context
 
     public async Task<int?> GetIdAsync(string username)
     {
-        return await Search(username).Select(x => x.Id).FirstOrDefaultAsync();
+        return await SearchQuery(username, false).Select(x => x.Id).FirstOrDefaultAsync();
     }
 
     /// <summary>
     /// Helper method for searching by username consistently
     /// </summary>
     /// <param name="username">The username to search for</param>
+    /// <param name="partialMatch">Whether or not we want to check for partial name matches on lookup.</param>
     /// <returns>A query that filters players by the username provided</returns>
-    private IQueryable<Player> Search(string username)
+    private IQueryable<Player> SearchQuery(string username, bool partialMatch)
     {
+        if (partialMatch)
+        {
+            if (username.Contains(' '))
+            {
+                // Look for users with either ' ' or '_' in the name - osu only uses one (i.e. "Red Pixel" cannot coexist with "Red_Pixel")
+                return _context.Players.Where(p =>
+                    p.Username != null
+                    && (
+                        EF.Functions.ILike(p.Username, $"%{username}%")
+                    )
+                );
+            }
```

Comment:
ngl, just scrap all of this. If a user is *searching* for someone, they need to handle the spaces and underscores themselves. Even the osu website cannot decipher `[ MILK_Jiang]` from `[_MILK_Jiang]`. Same for line 192, just scrap the if statement and return the ILike.

This code also doesn't do what I just described, it's a duplicate of what's below.